Libraries/LibJS/Runtime/Object.cpp

```diff
@@ -631,9 +631,12 @@ Value Object::delete_property(const PropertyName& property_name)
 
     if (property_name.is_number())
         return Value(m_indexed_properties.remove(property_name.as_number()));
-    int property_index = property_name.as_string().to_int().value_or(-1);
-    if (property_index >= 0)
-        return Value(m_indexed_properties.remove(property_name.as_number()));
+
+    if (property_name.is_string()) {
+        i32 property_index = property_name.as_string().to_int().value_or(-1);
```

Comment:
Hmm. Perhaps it would be better to check when constructing a `PropertyName` from a string if the string is in fact a number, and if so, turning into a numeric `PropertyName`. Then we wouldn't have to worry about this sort of logic when using a `PropertyName`.

What do you think?

Reply:
Nice idea, and I briefly tried it, but it made other things fall apart. Let's do this separately.